src/umpire/Allocator.hpp

```diff
@@ -64,6 +66,8 @@ class Allocator : private strategy::mixins::Inspector, strategy::mixins::Allocat
    */
   inline void* allocate(std::size_t bytes);
 
+  inline void* allocate(Resource const& r, std::size_t bytes);
```

Comment:
Can we consider switching the order of the arguments so that the size always comes first? I think that would make it easier to use all these overloads of allocate. Switching the argument order for deallocate would also be nice.

Reply:
We looked in to that implementation, but because of how the `allocate` function is already overloaded, plus the way that `named_allocate` was implemented, we decided to go with the resource (or the name in the case of `named_allocate`) always comes first. @davidbeckingsale may have more details...

Reply:
Yes, but what about named resource allocations?